Add importer to resources, format go imports

agadelshin opened the pull request:
Closes: #48 

PR to support resource imports with docs and fmt of golang imports

TimSimpson reviewed (changes requested):
When I create a small terraform project and try to import the project resource, I get an error saying that the remote state doesn't exist even when it should.

Here's my steps:

1. build this branch of the provider
2. copy `terraform-provider-eventstorecloud` to a new directory
3. add `main.tf` to that new directory with the following content:

```terraform
terraform {
  required_providers {
    eventstorecloud = {
    }
  }
}

resource "eventstorecloud_project" "project" {
  name = "My Project"
}

output "project_id" {
  value = eventstorecloud_project.project.id
}
```
4. run `terraform init`
5. run `terraform import eventstorecloud_project.project XYZ`, where `XYZ` is a valid project I can see in the console

The output is:
```
eventstorecloud_project.project: Importing from ID "XYZ"...
eventstorecloud_project.project: Import prepared!
  Prepared eventstorecloud_project for import
eventstorecloud_project.project: Refreshing state... [id=XYZ]

Error: Cannot import non-existent remote object

While attempting to import an existing object to
eventstorecloud_project.project, the provider detected that no object exists
with the given id. Only pre-existing objects can be imported; check that the
id is correct and that it is associated with the provider's configured region
or endpoint, or use "terraform apply" to create a new remote object for this
resource.
```

agadelshin commented:
Wasn’t able to reproduce it with project, but network resource had the issue you mentioned, so took a look how to fix it, it seems during the import project_id wasn’t set correctly. 

I checked how it’s solved in other providers and wrote a simple parser for {project_id}:{network_id} format. Will check other resources as well.

agadelshin commented:
Added helper function to parse {project_id}:{id} format, added everywhere except eventstorecloud_project resource.

tested resources:

* network
* managed cluster
* scheduled backups


not tested:
* integration
* peering

TimSimpson commented:
Ok, I made a mistake here. I was seeing the problem with scheduled backups from some back end tests I run, but when I went to write the issue up I used projects (which I picked because it was easier) and forget to set the org ID environment variable... classic mistake. :)

I think this is good to go, but I just want to take some time to test the other resources before I sign off.

agadelshin commented:
```
terraform import eventstorecloud_peering.test c5dfaoto0aembbqpbrqg:c5e4hhlo0aembvvdqtpg
eventstorecloud_peering.test: Importing from ID "c5dfaoto0aembbqpbrqg:c5e4hhlo0aembvvdqtpg"...
eventstorecloud_peering.test: Import prepared!
  Prepared eventstorecloud_peering for import
eventstorecloud_peering.test: Refreshing state... [id=c5e4hhlo0aembvvdqtpg]

Error: GCP peering link missing remote peering link project identifier



Error: GCP peering link missing remote peering link network identifier
```

I found an error with peering import, apparently we also need to set project/network ids for it.

TimSimpson commented on esc/import.go:
```diff
@@ -0,0 +1,33 @@
+	result := strings.Split(importId, ":")
+
+	if len(result) != 2 {
+		return nil, fmt.Errorf("failed to parse import id")
```
The user might get the import ID wrong initially so it could be helpful to include it in the error message:
```suggestion
		return nil, fmt.Errorf("Failed to parse import id. Please use the format `{project_id}:{resource_id}`")
```

```bash
$ terraform import eventstorecloud_network.network c5e4fbcbcv45qdammis0
eventstorecloud_network.network: Importing from ID "c5e4fbcbcv45qdammis0"...

Error: Failed to parse import id. Please use the format `{project_id}:{resource_id}`
```
